src/Symfony/Component/Messenger/Transport/Serialization/Serializer.php

```diff
@@ -70,9 +71,8 @@ public function decode(array $encodedEnvelope): Envelope
         $stamps = $this->decodeStamps($encodedEnvelope);
 
         $context = $this->context;
-        /** @var SerializerStamp|null $serializerStamp */
-        if ($serializerStamp = $stamps[SerializerStamp::class] ?? null) {
-            $context = $serializerStamp->getContext() + $context;
+        if (isset($stamps[SerializerStamp::class])) {
+            $context = end($stamps[SerializerStamp::class])->getContext() + $context;
```

Comment:
we can actually support multiple stamps no?

Reply:
like merging all stamps together? I don't know, so I wouldn't change that in this PR :)

Reply:
i think deciding on end() vs reset() vs merging kinda belongs to this PR

Reply:
about reset() vs end(): the previous behavior returned the latest stamp, so it should be env() - in need of any reason to change it I'd keep the behavior.